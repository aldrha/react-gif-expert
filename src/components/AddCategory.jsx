import { useState } from 'react';

export const AddCategory = ({ onNewCategory }) => {
	const [inputValue, setInputValue] = useState('');

	const onInputChange = ({ target }) => {
		setInputValue(target.value);
	};

	const onSubmit = (event) => {
		event.preventDefault();
		const valueFinal = inputValue.trim();

		if (valueFinal.length <= 1) return;

		onNewCategory(valueFinal);
		// setCategories((categories) => [inputValue, ...categories]);
		setInputValue('');
	};

	return (
		<form onSubmit={onSubmit}>
			{/* si el metodo solo recibe un evento [(event) => onSubmit(event), (event) => onInputChange(event)] ==> entonces se puede resumir con solo el nombre del mismo */}
			<input type="text" placeholder="Buscar gifs" value={inputValue} onChange={onInputChange} />
		</form>
	);
};
