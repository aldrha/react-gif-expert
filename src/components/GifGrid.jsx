import { GifItem } from './GifItem';
import { useFetchGifs } from '../hooks/useFetchGifs';
import PropTypes from 'prop-types';

export const GifGrid = ({ category }) => {
    const { images, isLoading } = useFetchGifs(category);

    return (
        <>
            <h1 aria-label="category-title">{category}</h1>
            {isLoading && <h2>Cargando ...</h2>}

            {/* Las propiedades se pueden desestructurar sino son muchas, cuando son muchas pueden enviarse como se hacen comun mente => img.id, img.title, img.url o usar la propiedad {...img} (esto significa esparcir las props y se usa cuando son bastantes) */}
            <div className="card-grid">
                {images.map((img) => (
                    <GifItem key={img.id} {...img} />
                ))}
            </div>
        </>
    );
};

GifGrid.propTypes = {
    category: PropTypes.string.isRequired,
};
