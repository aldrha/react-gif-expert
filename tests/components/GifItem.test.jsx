import { render, screen } from '@testing-library/react';
import { GifItem } from '../../src/components/GifItem';

describe('Pruebas en <GifItem />', () => {
	const title = 'Esto es un titulo';

	const url =
		'https://fastly.picsum.photos/id/360/536/354.jpg?hmac=hkJ_RmEPujjDj-tkqFkWT01yZlT-pWSePQb2PUJlwvE';

	test('Debe hacer match con el snapshop', () => {
		const { container } = render(<GifItem title={title} url={url} />);
		expect(container).toMatchSnapshot();
	});

	test('debe mostrar la imagen con el URL y el ALT indicado', () => {
		render(<GifItem title={title} url={url} />);
		// expect(screen.getByRole('img').src).toBe(img.url);
		// expect(screen.getByRole('img').alt).toBe(img.alt);
		const { src, alt } = screen.getByRole('img');
		expect(src).toBe(url);
		expect(alt).toBe(title);
	});

	test('debe de mostrar el titulo en el componente', () => {
		render(<GifItem title={title} url={url} />);
		expect(screen.getAllByText(title)).toBeTruthy();
	});
});
