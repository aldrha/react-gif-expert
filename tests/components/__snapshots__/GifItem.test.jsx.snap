// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Pruebas en <GifItem /> Debe hacer match con el snapshop 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Esto es un titulo"
      src="https://fastly.picsum.photos/id/360/536/354.jpg?hmac=hkJ_RmEPujjDj-tkqFkWT01yZlT-pWSePQb2PUJlwvE"
    />
    <p>
       
      Esto es un titulo
    </p>
  </div>
</div>
`;
